business/apps.go

```diff
@@ -279,3 +289,33 @@ func fetchNamespaceApps(layer *Layer, namespace string, appName string) (namespa
 
 	return allEntities, nil
 }
+
+type appServiceWithTracing struct {
```

Comment:
Ah, ok, so the "AOP" style is implemented providing a different implementation of the interface for tracing/no-tracing.
I think I understand it, I'd like to fetch it locally and review it closely, it looks correct, but as the amount of changes is high I'd review if there is no side effect.

Reply:
> Ah, ok, so the "AOP" style is implemented, providing a different interface implementation for tracing/no-tracing.

I'm not familiar with "AOP" but yeah that's the gist. Only the interface gets exposed and there's a separate implementation that has all the tracing logic that just wraps the original. In another language, something like a tracing decorator would probably be nice here:

Python example
```
def tracing(func):
    def wrapper():
        start_trace()
        func()
        end_trace()
    return wrapper

class AppService:
    @tracing
    def get_apps():
    ...
    @tracing
    def list_apps():
...
```

I'm definitely open to changing this and welcome feedback. I was worried about the amount of code changes involved with this approach but keeping the tracing boilerplate out of the business logic made the integration cleaner. I even played around a bit with code generation for the various tracing interfaces so that the tracing implementations wouldn't need to be maintained individually.